targets/TARGET_STM/rtc_api_hal.h

```diff
@@ -38,7 +38,26 @@
 extern "C" {
 #endif
 
-#if MBED_CONF_TARGET_LSE_AVAILABLE
+// Possible choices of the RTC_CLOCK_SOURCE configuration set in json file
+#define USE_RTC_CLK_LSE_OR_LSI 1
+#define USE_RTC_CLK_LSI 2
+#define USE_RTC_CLK_HSE 3
+
+#if !((MBED_CONF_TARGET_RTC_CLOCK_SOURCE == USE_RTC_CLK_LSE_OR_LSI) || (MBED_CONF_TARGET_RTC_CLOCK_SOURCE == USE_RTC_CLK_LSI) || (MBED_CONF_TARGET_RTC_CLOCK_SOURCE == USE_RTC_CLK_HSE))
+#error "RTC clock configuration is invalid!"
+#endif
+
+#if (MBED_CONF_TARGET_RTC_CLOCK_SOURCE == USE_RTC_CLK_HSE) && !(TARGET_STM32F2 || TARGET_STM32F3 || TARGET_STM32F4)
```

Comment:
> !(TARGET_STM32F2 || TARGET_STM32F3 || TARGET_STM32F4)

If I grep RCC_RTCCLKSOURCE_HSE_DIVX in the code, I could find: F2/F4/F7/L0/L1 ?

Reply:
L0/L1 have a different divider register configuration, which won't work with `(RCC_RTCCLKSOURCE_HSE_DIVX | RTC_HSE_DIV << 16)`

Reply:
F3 is wrong though, you're right. Gotta change that with F7.

Reply:
let's wait for complete review